.github/workflows/publish-snapshot.yml

```diff
@@ -9,19 +9,13 @@ jobs:
     steps:
       - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3
 
-      - name: Cache Gradle Files
-        uses: actions/cache@e12d46a63a90f2fae62d114769bbf2a179198b5c # v3
-        with:
-          path: |
-            ~/.gradle/caches/
-            ~/.gradle/wrapper/
-          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
-          restore-keys: |
-            ${{ runner.os }}-gradle-
+      - name: Setup gradle
+        uses: gradle/actions/setup-gradle@v3
+
       - name: Set up Java
         uses: actions/setup-java@cd89f46ac9d01407894225f350157564c9c7cee2 # v3
         with:
-          distribution: 'zulu'
+          distribution: 'temurin'
```

Comment:
The tests **before merge** should run on the same JDK that publishes, to prevent surprises when releasing, which is **after merge**. (i.e. add setup-java to the other workflow)

Reply:
I did that change just to align with the distribution that we use in the main repo. But this change it not related with this PR so I just reverted it.